zio-http/src/main/scala/zhttp/service/server/ServerChannelFactory.scala

```diff
@@ -15,13 +15,13 @@ object ServerChannelFactory {
 
   object Live {
```

Comment:
without the casts in this and ChannelFactory, I was getting 
```
zhttp.service.server.ServerChannelFactory.Live
[info] compiling 1 Scala source to /Users/brown/git/zio-http/zio-http/target/scala-3.0.0-RC1/classes ...
[error] -- [E007] Type Mismatch Error: /Users/brown/git/zio-http/zio-http/src/main/scala/zhttp/service/server/ServerChannelFactory.scala:18:16 
[error] 18 |      UIO(() => new JNioServerSocketChannel())//.asInstanceOf[UIO[JChannelFactory[JServerChannel]]]
[error]    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |    Found:    () => zhttp.core.JServerChannel
[error]    |    Required: io.netty.channel.ChannelFactory[zhttp.core.JServerChannel]
[error] -- [E007] Type Mismatch Error: /Users/brown/git/zio-http/zio-http/src/main/scala/zhttp/service/server/ServerChannelFactory.scala:21:16 
[error] 21 |      UIO(() => new JEpollServerSocketChannel())//.asInstanceOf[UIO[JChannelFactory[JServerChannel]]]
[error]    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |    Found:    () => zhttp.core.JServerChannel
[error]    |    Required: io.netty.channel.ChannelFactory[zhttp.core.JServerChannel]
[error] -- [E007] Type Mismatch Error: /Users/brown/git/zio-http/zio-http/src/main/scala/zhttp/service/server/ServerChannelFactory.scala:24:16 
[error] 24 |      UIO(() => new JKQueueServerSocketChannel())//.asInstanceOf[UIO[JChannelFactory[JServerChannel]]]
[error]    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[error]    |    Found:    () => zhttp.core.JServerChannel
[error]    |    Required: io.netty.channel.ChannelFactory[zhttp.core.JServerChannel]
```
when doing the scala 3 compile

Reply:
@tusharmath this is the issue I get when compiling under scala 3. I also need to to tweak the ci instructions so that the scala fix doesn't fail under scala 3 at the moment as well.